kraken-build/src/kraken/std/cargo/tasks/cargo_hack_task.py

```diff
@@ -0,0 +1,57 @@
+import subprocess
+from enum import Enum, auto
+
+from kraken.core import Property, Task, TaskStatus
+
+
+class CargoHackFeatures(Enum):
+    """Feature sets to test"""
+
+    EACH = auto()
+    POWERSET = auto()
+
+
+class CargoHackAction(Enum):
+    """CargoHackAction to run for every feature set"""
+
+    CHECK = auto()
+    TEST = auto()
+    BUILD = auto()
+
+
+class CargoHackTask(Task):
+    description = "Executes cargo hack to verify crate features."
+    error_message: Property[str | None] = Property.default(None)
+    features: Property[CargoHackFeatures] = Property.default(CargoHackFeatures.EACH)
+    action: Property[CargoHackAction] = Property.default(CargoHackAction.CHECK)
+    action_arguments: Property[list[str]] = Property.default_factory(list)
+
+    def execute(self) -> TaskStatus:
+        command = ["cargo", "hack"]
+
+        match self.features.get():
+            case CargoHackFeatures.EACH:
+                command.append("--each-feature")
+            case CargoHackFeatures.POWERSET:
+                command.append("--feature-powerset")
+
+        match self.action.get():
+            case CargoHackAction.CHECK:
+                command.append("check")
+            case CargoHackAction.TEST:
+                command.append("test")
+            case CargoHackAction.BUILD:
+                command.append("build")
+
+        for arg in self.action_arguments.get():
+            command.append(arg)
+
+        result = subprocess.run(command, cwd=self.project.directory)
+        if result.returncode == 0:
+            return TaskStatus.succeeded()
+
+        return self.error_message.map(
```

Comment:
Where do you expect `error_message` to be filled? From the user in `.kraken.py`? What would a possible error message be?

I lead on removing the property and just returning `TaskStatus.from_exit_code(command, result.returncode)`.

Reply:
Honestly, this code is copy paste from another task and I did not know what I'm trying to do :D will do what you suggest.